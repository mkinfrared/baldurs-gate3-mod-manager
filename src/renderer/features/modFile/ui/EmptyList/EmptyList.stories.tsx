import { Meta, StoryObj } from "@storybook/react";

import { EmptyList } from "./EmptyList";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/EmptyList",
  component: EmptyList,
  tags: ["autodocs"],
} satisfies Meta<typeof EmptyList>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <EmptyList {...args} />
    </div>
  ),
};

export { Default };

export default meta;
