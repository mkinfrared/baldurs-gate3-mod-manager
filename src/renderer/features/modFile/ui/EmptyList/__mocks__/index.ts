export { default } from "./EmptyList.mock";
