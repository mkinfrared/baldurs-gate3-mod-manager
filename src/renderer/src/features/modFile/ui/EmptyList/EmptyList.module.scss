@layer EmptyList, base;

.EmptyList {
  display: grid;
  flex: 1;
  place-content: center;

  @layer EmptyList {
    border-top: none;
  }

  .header {
    margin-bottom: 2rem;
  }

  > input {
    display: none;
  }
}
