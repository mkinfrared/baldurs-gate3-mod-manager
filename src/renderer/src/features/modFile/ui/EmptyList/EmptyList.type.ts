import { ChangeEventHandler } from "react";

export type EmptyListProps = {
  acceptedFileTypes: string[];
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  error?: string;
  onFilesSelect: ChangeEventHandler<HTMLInputElement>;
};
