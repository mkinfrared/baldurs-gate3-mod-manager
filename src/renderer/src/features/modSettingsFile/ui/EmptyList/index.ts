export { EmptyList } from "./EmptyList";

export * from "./EmptyList.type";
