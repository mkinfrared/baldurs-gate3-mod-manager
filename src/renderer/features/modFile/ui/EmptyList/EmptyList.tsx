import { classNames } from "renderer/shared/lib/helpers";
import { Button, Card, Heading } from "renderer/shared/ui";

import css from "./EmptyList.module.scss";
import { EmptyListProps } from "./EmptyList.type";

const EmptyList = ({ className }: EmptyListProps) => (
  <Card
    className={classNames(css.EmptyList, className)}
    data-testid="EmptyList"
  >
    <Heading variant="h4" className={css.header}>
      Drag and drop you zip or rar files here or
    </Heading>
    <Button type="button">
      <label htmlFor="file">Click this button to select files</label>
    </Button>
    <input type="file" id="file" />
  </Card>
);

export { EmptyList };
