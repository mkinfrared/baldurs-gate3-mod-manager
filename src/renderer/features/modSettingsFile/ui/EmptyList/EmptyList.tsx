import { classNames } from "renderer/shared/lib/helpers";
import { Card } from "renderer/shared/ui";

import css from "./EmptyList.module.scss";
import { EmptyListProps } from "./EmptyList.type";

const EmptyList = ({ className, children }: EmptyListProps) => (
  <Card
    className={classNames(css.EmptyList, className)}
    data-testid="EmptyList"
  >
    {children}
  </Card>
);

export { EmptyList };
