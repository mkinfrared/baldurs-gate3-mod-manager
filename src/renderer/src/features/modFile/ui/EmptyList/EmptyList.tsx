import { classNames } from "@renderer/shared/lib/helpers";
import { Button, Card, Heading } from "@renderer/shared/ui";

import css from "./EmptyList.module.scss";
import { EmptyListProps } from "./EmptyList.type";

const EmptyList = ({
  acceptedFileTypes,
  className,
  onFilesSelect,
}: EmptyListProps) => (
  <Card
    className={classNames(css.EmptyList, className)}
    data-testid="EmptyList"
  >
    <Heading variant="h4" className={css.header}>
      Drag and drop you {acceptedFileTypes.join(", ")} files here or
    </Heading>
    <Button type="button">
      <label htmlFor="file">Click this button to select files</label>
    </Button>
    <input
      type="file"
      id="file"
      multiple
      accept={acceptedFileTypes.join(",")}
      onChange={onFilesSelect}
    />
  </Card>
);

export { EmptyList };
