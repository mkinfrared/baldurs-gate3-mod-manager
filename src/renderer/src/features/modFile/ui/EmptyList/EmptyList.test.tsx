import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { EmptyList } from "./EmptyList";

describe("<EmptyList />", () => {
  const onFileSelect = vi.fn();
  const acceptedFileTypes = ["zip", "rar"];

  const Component = (
    <EmptyList
      acceptedFileTypes={acceptedFileTypes}
      onFilesSelect={onFileSelect}
    />
  );

  it("should be defined", () => {
    expect(EmptyList).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
