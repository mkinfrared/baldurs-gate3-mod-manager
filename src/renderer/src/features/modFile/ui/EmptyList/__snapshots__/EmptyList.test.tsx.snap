// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<EmptyList /> > should match the snapshot 1`] = `
<div>
  <div
    class="Card EmptyList"
    data-testid="Card"
  >
    <h4
      class="Heading header"
      data-testid="Heading"
    >
      Drag and drop you 
      zip, rar
       files here or
    </h4>
    <button
      class="Button contained primary"
      data-testid="Button"
      type="button"
    >
      <svg
        class="Spinner spinner"
      >
        bars.svg
      </svg>
      <span
        class="content"
      >
        <label
          for="file"
        >
          Click this button to select files
        </label>
      </span>
    </button>
    <input
      accept="zip,rar"
      id="file"
      multiple=""
      type="file"
    />
  </div>
</div>
`;
